### PR TITLE
Helm chart: mount RMS webstudio & security folders based on single flag

### PR DESCRIPTION
**Pull request by rameshpolishetti**

### CODE REVIEW

**Comment by rameshpolishetti** [comment]
@LakshmiMekala validate the PR.

File: cloud/kubernetes/helm/templates/_helpers.tpl [hunk 1]
```diff
@@ -29,11 +29,9 @@ volumeMounts:
   mountPath: "/opt/tibco/be/{{ .Values.beShortVersion }}/rms/shared"
 {{- end }}
 {{- if .Values.rmsDeployment }}
-{{- if .Values.persistence.rmsSecurity }}
```
Comment (LakshmiMekala) [inline cloud/kubernetes/helm/templates/_helpers.tpl:32]: We can delete this if condition for security and content to webstudion

File: cloud/kubernetes/helm/templates/_helpers.tpl [hunk 2]
```diff
@@ -45,7 +43,7 @@ volumeMounts:
 volumes:
 {{- end }}
 {{- range $i, $vName := tuple "data-store" "logs" "rms-shared" "rms-security" "rms-webstudio" }}
-{{- if or (and (eq $vName "data-store") (eq $.Values.bsType "sharednothing")) (and (eq $vName "logs") $.Values.persistence.logs) (and (eq $vName "rms-shared") (or $.Values.enableRMS $.Values.rmsDeployment)) (and (eq $vName "rms-security") $.Values.persistence.rmsSecurity $.Values.rmsDeployment) (and (eq $vName "rms-webstudio") $.Values.persistence.rmsWebstudio $.Values.rmsDeployment) }}
```
Comment (LakshmiMekala) [inline cloud/kubernetes/helm/templates/_helpers.tpl:48]: `$.Values.persistence.rmsWebstudio $.Values.rmsDeployment)` is repeated twice and rms-security condition is also there

File: cloud/kubernetes/helm/templates/_helpers.tpl [hunk 3]
```diff
@@ -45,7 +43,7 @@ volumeMounts:
 volumes:
 {{- end }}
 {{- range $i, $vName := tuple "data-store" "logs" "rms-shared" "rms-security" "rms-webstudio" }}
```
Comment (LakshmiMekala) [inline cloud/kubernetes/helm/templates/_helpers.tpl:45]: can we delete rms-security?

Also the same range tuples with same values were added in many places, can we move the range values to values.yaml

File: cloud/kubernetes/helm/templates/pvc.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {{- if .Values.persistence.scSupportsReadWriteMany }}
 {{- range $i, $vName := tuple "data-store" "logs" "rms-shared" "rms-security" "rms-webstudio" }}
-{{- if or (and (eq $vName "data-store") (eq $.Values.bsType "sharednothing")) (and (eq $vName "logs") $.Values.persistence.logs) (and (eq $vName "rms-shared") $.Values.rmsDeployment) (and (eq $vName "rms-security") $.Values.persistence.rmsSecurity $.Values.rmsDeployment) (and (eq $vName "rms-webstudio") $.Values.persistence.rmsWebstudio $.Values.rmsDeployment) }}
```
Comment (LakshmiMekala) [inline cloud/kubernetes/helm/templates/pvc.yaml:3]: $.Values.persistence.rmsWebstudio $.Values.rmsDeployment) is repeated twice and rms-security condition is also there

File: cloud/kubernetes/helm/templates/pv-manifest.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {{- if and (eq .Values.persistence.storageClass "-") (or (eq .Values.cpType "minikube") (eq .Values.cpType "aws")) }}
 {{- range $i, $vName := tuple "data-store" "logs" "rms-shared" "rms-security" "rms-webstudio" }}
-{{- if or (and (eq $vName "data-store") (eq $.Values.bsType "sharednothing")) (and (eq $vName "logs") $.Values.persistence.logs) (and (eq $vName "rms-shared") $.Values.rmsDeployment) (and (eq $vName "rms-security") $.Values.persistence.rmsSecurity $.Values.rmsDeployment) (and (eq $vName "rms-webstudio") $.Values.persistence.rmsWebstudio $.Values.rmsDeployment) }}
```
Comment (LakshmiMekala) [inline cloud/kubernetes/helm/templates/pv-manifest.yaml:3]: $.Values.persistence.rmsWebstudio $.Values.rmsDeployment) is repeated twice and rms-security condition is also there

**Review by LakshmiMekala — CHANGES_REQUESTED** [review]
Please review my comments

File: cloud/kubernetes/helm/templates/_helpers.tpl (hunk 1 above)
Comment (rameshpolishetti) [reply cloud/kubernetes/helm/templates/_helpers.tpl:32]: Yes this what the commit does.

File: cloud/kubernetes/helm/templates/_helpers.tpl (hunk 2 above)
Comment (rameshpolishetti) [reply cloud/kubernetes/helm/templates/_helpers.tpl:48]: Here `rms-security` is loop variable check. Condition repeats for `rms-webstudio` as well.

File: cloud/kubernetes/helm/templates/_helpers.tpl (hunk 3 above)
Comment (rameshpolishetti) [reply cloud/kubernetes/helm/templates/_helpers.tpl:45]: NO. rms-security folder need to be mounted.

Values.yaml should be used to collect flags/data from the user while installing the chart. Not a good idea to move this to values.yaml

File: cloud/kubernetes/helm/templates/pv-manifest.yaml (hunk 5 above)
Comment (rameshpolishetti) [reply cloud/kubernetes/helm/templates/pv-manifest.yaml:3]: Here rms-security is loop variable check. Condition repeats for rms-webstudio as well.

**Comment by LakshmiMekala** [comment]
Validated PR, LGTM

**Review by LakshmiMekala — APPROVED** [review]
LGTM

**Pull request merged**